clean up programs etc related to obsolete perl re-gridding package (regrid.pl)

gmao-rreichle opened the pull request:
Cleans up programs etc from the obsolete perl package for re-gridding restarts. 

Changed base branch to develop (30-Jun-2026).

Trivially 0-diff for running the GCM.  Zero-diff for make_bcs and remap_restarts.

Related PRs:
https://github.com/GEOS-ESM/GEOS_Util/pull/113   (was merged into "v12 GCM" branch before the merger of the v12 branch into develop)

cc: @mathomp4 @sdrabenh @weiyuan-jiang @biljanaorescanin

gmao-rreichle committed:
removed obsolete program used by regrid.pl (mk_GEOSldasRestarts.F90)

weiyuan-jiang committed:
remove more files not needed

gmao-rreichle committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl

gmao-rreichle committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl

gmao-rreichle committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl

gmao-rreichle committed:
removed additional obsolete "regrid" files

gmao-rreichle committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl (manually resolved conflicts by re-deleting obsolete files mk_CatchCNRestarts.F90, mk_CatchRestarts.F90, mk_GEOSldasRestarts.F90)

gmao-rreichle committed:
removed names of obsolete files from CMakeLists.txt

gmao-rreichle committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl

mathomp4 committed:
Merge branch 'feature/sdrabenh/gcm_v12' into feature/rreichle/remove-regridpl

mathomp4 commented:
@gmao-rreichle I fixed up the spack CI issue. You must have made this PR in the like 10 minutes when I fixed that up in v12. It should be happy.

gmao-rreichle commented:
> @gmao-rreichle I fixed up the spack CI issue. You must have made this PR in the like 10 minutes when I fixed that up in v12. It should be happy.

Thanks!  Should be good to go, then.  I'll change to "ready for review".

mathomp4 commented:
I've labeled as zero-diff because, well, it sort of has to be :)

mathomp4 committed:
Merge branch 'feature/sdrabenh/gcm_v12' into feature/rreichle/remove-regridpl

# Conflicts:
#	.github/workflows/spack-ci.yml

mathomp4 commented:
@gmao-rreichle Please take a look and make sure my branch updating didn't break anything. it looks good to me

sdrabenh committed:
Merge branch 'feature/sdrabenh/gcm_v12' into feature/rreichle/remove-regridpl

gmao-rreichle committed:
Merge branch 'feature/sdrabenh/gcm_v12' into feature/rreichle/remove-regridpl

gmao-rreichle committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl

gmao-rreichle commented on GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/mk_restarts/CMakeLists.txt:
```diff
@@ -7,14 +7,10 @@ set(srcs
-  mk_CatchCNRestarts.F90
-  mk_CatchRestarts.F90
   mk_LakeLandiceSaltRestarts.F90
   mk_GEOSldasRestarts.F90
```
This PR deletes `mk_GEOSldasRestarts.F90`.  I suppose it needs to be removed from CMakeLists.txt?

mathomp4 replied:
Oh. Yes that is true. CMake will not be happy otherwise

gmao-rreichle replied:
I removed mk_GEOSldasRestarts.F90 from CMakeLists.txt

gmao-rreichle committed:
Remove mk_GEOSldasRestarts.F90 from CMakeLists.txt

Removed mk_GEOSldasRestarts.F90 from the list of source files.

gmao-rreichle commented on GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Utils/Raster/makebcs/mod_process_hres_data.F90:
```diff
@@ -31,10 +31,6 @@ MODULE process_hres_data
-#if defined USE_EXTERNAL_FINDLOC
-  use findloc_mod,            only: findloc
-#endif
-
```
@mathomp4 : @weiyuan-jiang and I _think_ we can delete findloc(), but a few years ago you made the following commit: https://github.com/GEOS-ESM/GEOSgcm_GridComp/commit/17829f500426b49cbb90616367cc61718302910b
Should we keep findloc() or can it be deleted?

mathomp4 replied:
@gmao-rreichle It's probably safe to remove. I'd be extremely disappointed if any compiler we currently use doesn't support `findloc()` anymore.

From what I can test on godbolt and on bucy, everything we might build GEOS with handles it just fine.

Would you like one of us to remove it?

gmao-rreichle replied:
Thanks, @mathomp4.  The removal was already committed to the branch of this PR by @weiyuan-jiang, so the PR is ready for final testing and merging.  I can't formally approve the PR because I created it.  It'll need CMake team approval. 

When @biljanaorescanin is back, she can test it and forward it to @sdrabenh.

mathomp4 replied:
@gmao-rreichle Oh. okay. I've approved for CMake but I now realize that means I approved for everyone. Sorry, just approved by habit. But I won't merge, I'll wait for @sdrabenh and @biljanaorescanin to test.

sdrabenh committed:
Merge branch 'develop' into feature/rreichle/remove-regridpl
